azure-iot-provisioning-devicessdk/azure/iot/provisioning/devicessdk/device/registration_client.py

```diff
@@ -0,0 +1,43 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""
+This module provides an abstract interface representing clients which can communicate with the
+Device Provisioning Service.
+"""
+
+import abc
+import six
+
+
+@six.add_metaclass(abc.ABCMeta)
+class RegistrationClient(object):
+    """
+    Super class for any client that can be used to register devices to Device Provisioning Service.
+    :ivar str provisioning_host: Host running the Device Provisioning Service.
+    :ivar str security_client: Instance of Security client
+    :ivar str module_id: Instance of transport.
```

Comment:
None of these exist in the abstract class.

And the ivar that DOES exist (transport) isn't listed.

Reply:
changed to private